### PR TITLE
Restructure internals of library to use @storybook/store logic

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What Changed

The @storybook/testing-* libraries have lots of redundant logic that don't follow the development speed of Storybook. This PR aims to reuse the logic from Storybook's `prepareStory` instead of reinventing the wheel.

## Checklist

These are things to do before merging this:

[ ] expose normalizeStories + normalizeComponentAnnotations from @storybook/store
[ ] instead of using processCSFFile, use normalizeStories + normalizeComponentAnnotations from @storybook/store
[ ] figure out why using render from @storybook/react throws `Cannot find module 'react-dom' from 'render.js'`

Check the ones applicable to your change:

- [x] Tests are updated
- [ ] Documentation is updated

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [ ] `patch`
- [x] `minor`
- [ ] `major`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -65,98 +79,50 @@ export function composeStory<GenericArgs>(
...
+  } = processCSFFile({ default: meta, ...{[story.storyName!]: story} }, '', meta.title)
+  const normalizedStory = Object.values(normalizedStories)[0] as any
+
+  const preparedStory = prepareStory(
+    normalizedStory,
+    normalizedMeta,
+    projectAnnotations
   );
 
   const defaultGlobals = Object.entries(
```
Comment (yannbf) [inline src/index.ts:102]: Hey @tmeasday I'm sure there's an existing way to compose the globalTypes in Storybook. Do you know where? Something related with the urlStore?

File: src/index.ts [hunk 2]
```diff
@@ -65,98 +79,50 @@ export function composeStory<GenericArgs>(
...
-  const combinedDecorators = [
-    ...(story.decorators || []),
-    ...(meta?.decorators || []),
-    ...(globalConfig.decorators || []),
-  ];
-
-  const decorated = defaultDecorateStory<ReactFramework>(
-    finalStoryFn as any,
-    combinedDecorators as any
+  const projectAnnotations = { ...defaultGlobalConfig, ...globalConfig }
```
Comment (tmeasday) [inline src/index.ts:86]: PS we have `composeConfig` that does a little bit more than just `Object.assign` if that feels appropriate.

File: src/index.ts (hunk 1 above)
Comment (tmeasday) [reply src/index.ts:102]: Here! https://github.com/storybookjs/storybook/blob/4e007aa29b89d7cfd1da03479b6407ff008ccedb/lib/store/src/GlobalsStore.ts#L22-L37

File: src/index.ts (hunk 2 above)
Comment (yannbf) [reply src/index.ts:86]: Thanks! replaced the logic with `composeConfigs`

**Comment by yannbf** [comment]
Closing this as it was an experimentation-only PR

**Pull request closed without merging**